conda_build/metadata.py

```diff
@@ -148,9 +148,12 @@ def parse(data):
     for field in FIELDS:
         if field not in res:
             continue
+        # ensure that empty fields are dicts (otherwise selectors can cause invalid fields)
+        if not res[field]:
+            res[field] = {}
```

Comment:
I have added above three lines from 1.20.0 to fix a regression introduced in 1.20.1. Without these, the following code in a meta.yaml would lead to an error:
```
build:
    skip: True # [osx]
```

Because build is turned into a string.